src/Juvix/Compiler/Tree/Translation/FromCore.hs

```diff
@@ -0,0 +1,348 @@
+module Juvix.Compiler.Tree.Translation.FromCore where
+
+import Data.HashMap.Strict qualified as HashMap
+import Juvix.Compiler.Core.Data.BinderList qualified as BL
+import Juvix.Compiler.Core.Data.Stripped.InfoTable qualified as Core
+import Juvix.Compiler.Core.Language.Stripped qualified as Core
+import Juvix.Compiler.Tree.Data.InfoTable
+import Juvix.Compiler.Tree.Extra.Type
+import Juvix.Compiler.Tree.Language
+
+type BinderList = BL.BinderList
+
+fromCore :: Core.InfoTable -> InfoTable
+fromCore tab =
+  InfoTable
+    { _infoMainFunction = tab ^. Core.infoMain,
+      _infoFunctions = genCode tab <$> tab ^. Core.infoFunctions,
+      _infoInductives = translateInductiveInfo <$> tab ^. Core.infoInductives,
+      _infoConstrs = translateConstructorInfo <$> tab ^. Core.infoConstructors
+    }
+
+-- Generate code for a single function.
+genCode :: Core.InfoTable -> Core.FunctionInfo -> FunctionInfo
+genCode infoTable fi =
+  let argnames = map (Just . (^. Core.argumentName)) (fi ^. Core.functionArgsInfo)
+      code =
+        go
+          0
+          ( BL.fromList $
+              reverse
+                ( map
+                    (DRef . ArgRef)
+                    ( zipWithExact
+                        OffsetRef
+                        [0 .. fi ^. Core.functionArgsNum - 1]
+                        argnames
+                    )
+                )
+          )
+          (fi ^. Core.functionBody)
```

Comment:
I think it is possible to fuse the map with the zipWithExact: `zipWithExact (DRef . ArgRef . OffsetRef)`.

Could the formatting be improved somehow?

Reply:
Not exactly, because `OffsetRef takes two arguments`. But I improved the formatting.